core/src/main/java/org/infinispan/configuration/cache/IndexingConfiguration.java

```diff
@@ -20,12 +19,7 @@
    public static final AttributeDefinition<Boolean> AUTO_CONFIG = AttributeDefinition.builder("autoConfig", false).immutable().build();
    public static final AttributeDefinition<Set<Class<?>>> INDEXED_ENTITIES = AttributeDefinition.builder("indexed-entities", null, (Class<Set<Class<?>>>) (Class<?>) Set.class)
          .copier(CollectionAttributeCopier.INSTANCE)
-         .initializer(new AttributeInitializer<Set<Class<?>>>() {
-            @Override
-            public Set<Class<?>> initialize() {
-               return new HashSet<>(5);
-            }
-         }).immutable().build();
+         .initializer(() -> new HashSet<>(5)).immutable().build();
```

Comment:
I think it's better to use the default constructor now, because it doesn't allocate an array at all

Reply:
indeed